auth/simple_token.go

```diff
@@ -42,16 +42,18 @@ var (
 
 type simpleTokenTTLKeeper struct {
 	tokens          map[string]time.Time
-	stopCh          chan chan struct{}
+	donec           chan struct{}
+	stopc           chan struct{}
 	deleteTokenFunc func(string)
 	mu              *sync.Mutex
 }
 
 func (tm *simpleTokenTTLKeeper) stop() {
-	waitCh := make(chan struct{})
-	tm.stopCh <- waitCh
-	<-waitCh
-	close(tm.stopCh)
+	select {
+	case tm.stopc <- struct{}{}:
+	case <-tm.donec:
```

Comment:
So you mean the `stop()` function has to be synchronous? I am not sure I understand why it can't be asynchronous, since `t.simpleTokenKeeper` has been set to be `nil` before calling `stop()`.